api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -555,46 +557,71 @@ public DbUser syncComplianceTrainingStatus(DbUser dbUser)
       return dbUser;
     }
 
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
     try {
       Integer moodleId = dbUser.getMoodleId();
+      String email = dbUser.getUsername();
       if (moodleId == null) {
-        moodleId = complianceService.getMoodleId(dbUser.getUsername());
+        moodleId = complianceService.getMoodleId(email);
         if (moodleId == null) {
           // User has not yet created/logged into MOODLE
           return dbUser;
         }
         dbUser.setMoodleId(moodleId);
       }
 
-      List<BadgeDetails> badgeResponse = complianceService.getUserBadge(moodleId);
-      // The assumption here is that the User will always get 1 badge which will be AoU
-      if (badgeResponse != null && badgeResponse.size() > 0) {
-        BadgeDetails badge = badgeResponse.get(0);
-        Timestamp badgeExpiration =
-            badge.getDateexpire() == null
-                ? null
-                : new Timestamp(Long.parseLong(badge.getDateexpire()));
+      String expiryEpoch = null;
```

Comment:
This function is over a hundred lines long now. Can we split it into 4-5 pieces? I generally consider 20-line functions in need of a haircut.